Dockerfile

```diff
@@ -31,21 +37,21 @@ RUN apt-get install -y nodejs
 
 RUN mkdir -p /opt/QuickAnnotator
 WORKDIR /opt/QuickAnnotator
-COPY . /opt/QuickAnnotator
+COPY ./pyproject.toml /opt/QuickAnnotator/pyproject.toml
 
-ENV PATH="/opt/venv/bin:$PATH"
+# Install uv
+RUN pip install uv
 
-RUN python3 -m venv /opt/venv \
-    && /opt/venv/bin/python -m pip install --upgrade pip \
-    && /opt/venv/bin/pip install -e .
-
-# Install development python dependencies
-RUN /opt/venv/bin/pip install tqdm \
-                            ipykernel
+# Create a virtual environment for uv and install all dependencies
+RUN uv venv /opt/uv_venv
+ENV PATH="/opt/uv_venv/bin:$PATH"
+RUN uv pip install -r <(uv pip compile pyproject.toml)
 
 # Install node dependencies
+COPY ./quickannotator/client/package.json /opt/QuickAnnotator/quickannotator/client/package.json
+COPY ./quickannotator/client/package-lock.json /opt/QuickAnnotator/quickannotator/client/package-lock.json
 WORKDIR /opt/
-RUN npm install QuickAnnotator/quickannotator/client
+RUN npm install QuickAnnotator/quickannotator/client/
```

Comment:
does "npm ci" save us some speed?

Reply:
It does give us a speed boost - oddly npm ci is different from npm install in that it does not allow for installing node_modules in a directory other than the one containing package.json and package-lock.json.

I think this is still okay, I'll just have to copy these json files to /opt before installing.

Reply:
nnnnnnice. whats the approx timing difference?

Reply:
Down to ~7 seconds, although I'm suspicious there's caching involved. 

I'm not sure how cacheing would be possible though because I deleted the image which should prevent the use of cached layers.